pkg/apis/core/v1/resource_genclient.go

```diff
@@ -95,30 +95,21 @@ func (rwt ResourceWithTag) EndpointURL(ctx context.Context) (*url.URL, error) {
 	return url.Parse(fmt.Sprintf("/api/core/v1/resource.json/%v/tags/%v", rwt.Identifier, rwt.Tag))
 }
 
-func (rwt ResourceWithTag) FilterAPIRequest(ctx context.Context, req *http.Request) (*http.Request, error) {
-	endpointURL, err := rwt.EndpointURL(ctx)
+func (rwt ResourceWithTag) FilterRequestURL(ctx context.Context, url *url.URL) (*url.URL, error) {
+	op, err := types.OperationFromContext(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	baseURL := strings.TrimSuffix(req.URL.String(), endpointURL.String())
-
-	url := fmt.Sprintf("%v/api/core/v1/resource.json/%v/tags/%v", baseURL, rwt.Identifier, rwt.Tag)
-
-	newRequest, err := http.NewRequest(req.Method, url, nil)
-	if err != nil {
-		return nil, fmt.Errorf("error creating modified Request: %w", err)
+	if op == types.OperationDestroy {
+		url.Path = path.Dir(url.Path)
 	}
 
-	for h, vs := range req.Header {
-		if h != "Content-Type" && h != "Content-Length" {
-			for _, v := range vs {
-				newRequest.Header.Add(h, v)
-			}
-		}
-	}
+	return url, nil
+}
 
-	return newRequest, nil
+func (rwt ResourceWithTag) FilterAPIRequestBody(ctx context.Context) (interface{}, error) {
+	return "", nil
```

Comment:
this sends a response body containing just a JSON-encoded empty string - better return nil to prevent sending a response body (hope generic client already handles nil like that)

Reply:
details in code comments